src/lib/drivers/TSYS01.cpp

```diff
@@ -52,61 +49,61 @@ bool TSYS01::begin( float offsetCentiDegC, uint32_t signature, uint32_t caltime
   return rval;
 }
 
-void TSYS01::reset( void ) {
-  doCommand(RESET, RESET_WAIT_TIME_MS);
-}
+void TSYS01::reset(void) { doCommand(RESET, RESET_WAIT_TIME_MS); }
 
-bool TSYS01::getTemperature( float &temperature ) {
+bool TSYS01::getTemperature(float &temperature) {
   float temp = 0;
-  uint8_t data[3] = { 0 };
+  uint8_t data[3] = {0};
   bool spiTransactionSuccess = true;
-  bool rval = true;
-
-  spiTransactionSuccess &= doCommand(START_ADC_TEMP_CONV, ADC_CONV_WAIT_TIME_MS);
-  spiTransactionSuccess &= readData(READ_ADC_TEMP, data, sizeof(data));
+  bool rval = false;
 
-  // round() macro returns a long (4 bytes)
-  uint32_t adc16 = round(((float) (((uint32_t) data[0]) << 16 | ((uint32_t) data[1]) << 8 | data[2])) / 256);
-  for( uint8_t i = 0; i < CALIB_CNT; i++){
-    temp += _calibrations[i] * pow(adc16, 4 - i);
-  }
+  do {
+    // For SPI we need a way to check if the device is present.
+    if (!validatePROM()) {
```

Comment:
In the I2C version of the driver, we can rely on the I2c protocol to fail if a device isn't present (bc of the addressing scheme). For SPI, we need to add a validity check to make sure our device is responsive and valid. validateProm is good because it does a checksum computation.